redfish-core/lib/ethernet.hpp

```diff
@@ -2001,6 +2072,20 @@ inline void requestEthernetInterfacesRoutes(App& app)
 
             if (ipv4StaticAddresses)
             {
+                // TODO this is workaround needed to avoid ipv4 static and DHCP
+                // address co-existence This behaviour should be fixed in
+                // backend networkd D-bus application
+                bool ipv4Active =
+                    translateDhcpEnabledToBool(ethData.dhcpEnabled, true);
+                bool ipv6Active =
+                    translateDhcpEnabledToBool(ethData.dhcpEnabled, false);
+                if (ipv4Active)
+                {
+                    // Disable DHCPv4 while configuring ipv4 static addresses
+                    setDHCPEnabled(ifaceId, "DHCPEnabled", false, ipv6Active,
+                                   asyncResp);
+                }
```

Comment:
Can this be moved into createIPv4 method? So that some more error checks are passed? Just before creating the static IP object, we can disable the DHCPv4

Reply:
can be moved into createIPv4 method, but need to change createIPv4 input args.
 and createIPv4 method which was getting used in other places..

Reply:
done